src/odemis/acq/test/flim_test.py

```diff
@@ -0,0 +1,205 @@
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+
+import logging
+import odemis
+from odemis import model
+import time
+import os
+import unittest
+from unittest.case import skip
+from odemis.util import test
+from odemis.acq import stream
+
+logging.getLogger().setLevel(logging.DEBUG)
+logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)s:%(lineno)d %(message)s")
+
+CONFIG_PATH = os.path.dirname(odemis.__file__) + "/../../install/linux/usr/share/odemis/"
+SECOM_FLIM_CONFIG = CONFIG_PATH + "sim/secom-sim-flim.odm.yaml"
+
+'''
+To run these tests, you must run the Symphotime simulator.
+
+The simulator can be found at https://github.com/PicoQuant/SymPhoTime_Remote_Interface
+The PQ_ServerSimulator application is a Win32 or x64 based exe that simulates a Symphotime server
+By default, it listens on 127.0.0.1:6000 for a connection. Just launch the EXE with default settings
+in order to run these tests.
+'''
+
+
+class TestFlimtatic(unittest.TestCase):
+    backend_was_running = False
+
+    @classmethod
+    def setUpClass(cls):
+
+        try:
+            test.start_backend(SECOM_FLIM_CONFIG)
+        except LookupError:
+            logging.info("A running backend is already found, skipping tests")
+            cls.backend_was_running = True
+            return
+        except IOError as exp:
+            logging.error(str(exp))
+            raise
+
+        # Find CCD & SEM components
+        cls.sft = model.getComponent(role="time-correlator")
+        cls.tc_scanner = model.getComponent(role="tc-scanner")
+        cls.ex_light = model.getComponent(role="light")
+        cls.lscanner = model.getComponent(role="laser-mirror")
+        cls.apd = model.getComponent(role="tc-detector")
+        cls.detector = model.getComponent(role="photo-detector0")
+
+        cls.ex_light.period.value = 0.0001
+        em = [0] * len(cls.ex_light.emissions.value)
+        cls.ex_light.emissions.value = em
+        cls.ex_light.power.value = 0.25
+
+    @classmethod
+    def tearDownClass(cls):
+        time.sleep(1.0)
+
+        if cls.backend_was_running:
+            return
+        test.stop_backend()
+
+    def _on_image(self, im):
+        # Is called when an image is received in the live setting stream
+        logging.info("image: dim %s" , im.shape)
+        self._image = im
+
+    def validate_scan_sim(self, remote, repetition):
+        self.assertIsNotNone(remote.raw)
+        self.assertIsNotNone(remote.raw[0].metadata)
+        
+    def validate_image(self, img):
```

Comment:
Similar checks should go here, as for the raw. Check that the data is 3D (ie, RGB) and uint8. Check that the metadata fits the expectations.